lib/game.rb

```diff
@@ -0,0 +1,98 @@
+require 'pry'
+class Game
+  attr_accessor :board, :player_1, :player_2, :winning_token
+  include Players
+
+  #all win combos
+  WIN_COMBINATIONS = [
+    [0,1,2], #top row
+    [3,4,5], #middle row
+    [6,7,8], #bottom row
+    [0,3,6], #1st column
+    [1,4,7], #2nd column
+    [2,5,8], #third column
+    [0,4,8], #1 diagonal
+    [2,4,6] #2nd diagonal
+  ]
+
+  #initialize game with default human players and board if no arguments are given
+  def initialize(player_1 = Human.new("X"), player_2 = Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  #returns current player based on the turn count
+  def current_player
+    @board.turn_count.even? ? @player_1 : @player_2
+  end
+
+  #checks if game has been won
+  def won?
+    #iterate throguh the combination to check if either X or O is the winner
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```